src/core/Form/TextInput/TextInput.baseStyles.tsx

```diff
@@ -44,6 +45,26 @@ export const baseStyles = withSuomifiTheme(
       box-shadow: ${theme.shadows.actionElementBoxShadow};
     }
     }
+    
+  &.fi-text-input_with-icon {
+    & .fi-text-input_container {
+    position: relative;
+    }
+
+    & .fi-text-input_input{
+    padding-right: ${math(
+      `${theme.spacing.insetXl} * 2 + ${theme.spacing.insetM}`,
+    )};
+    }
+
+    & .fi-icon{
+        position: absolute;
+        width: 18px;
+        height: 18px;
+        top: ${theme.spacing.insetL};
+        right: ${theme.spacing.insetL};
```

Comment:
Same as above. In my opinion, we should stick with inset tokens when defining stuff inside the components.